### PR TITLE
overhaul API and audio demo, remove levelDB, add results page w/ client accuracy

### PR DESCRIPTION
**Pull request by asross**
This is a working but work-in-progress PR which overhauls the API we expose.

The biggest change is removing LevelDB (and the JSON format for tensors). In my experience with the deployed version of the audio demo, we had really huge overheads from saving/loading/aggregating ~30MB JSON versions of the weights, which made doing a responsive demo almost untenable. Sorry for introducing that :sweat_smile: 

Instead, as of this PR, we store copies of the updated weights in memory (in ArrayBuffer-serialized form). When we receive enough updated weights, we average them together and update the model. We save the model to disk using [`tf.Model.save`](https://js.tensorflow.org/api/0.12.0/#tf.Model.save), which has its own nicely-portable JSON manifest + binary weight file format. On startup, we just call `tf.loadModel` with either an `https://` URL (if no model has been saved) or a `file://` URL (if at least one has). 

We also remove data-saving from the `src/server` API entirely and move it to the audio demo. Instead of having separate socket and HTTP endpoints for saving tensors and files, we just have simple HTTP `POST /data` and `GET /data` endpoints. We save audio inputs to `.wav` files on disk and client/prediction metadata to JSON. Although this JSON metadata is backed up on disk / loaded on startup, while the server is running we just store it in a giant in-memory array. Since the metadata objects are small this shouldn't be an issue for the demo. We add a results page where we show accuracy/cross-entropy for different clients across model versions.

I'm definitely totally breaking the emoji demo, and I think I might need to add back in hyperparams, but let me know what you think!

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/federated-learning/27)

### CODE REVIEW

**Comment by aman-tiwari** [comment]
___
*[src/common/index.ts, line 228 at r1](https://reviewable.io/reviews/pair-code/federated-learning/27#-LIIEtxI9VvqsOidxqPr:-LIIGwiOAyUfIZgGmJTo:b-g5tx51) ([raw file](https://github.com/pair-code/federated-learning/blob/fb1f20d39cb50fa13d1961aa4b583965363ea15f/src/common/index.ts#L228)):*
<details><summary><i>Previously, asross (Andrew Ross) wrote…</i></summary><blockquote>

I'm definitely breaking the emoji demo here.
</blockquote></details>

Yup, but easily fixable if you have the model a complementary `load(...)` to load whatever format it saves itself in.

**Comment by aman-tiwari** [comment]
___
*[demo/audio/client/results.js, line 32 at r3](https://reviewable.io/reviews/pair-code/federated-learning/27#-LIMpD9TCq4zrfVLeoDI:-LINQ2Lo6-CMdaf7TgCI:b-h4ikto) ([raw file](https://github.com/pair-code/federated-learning/blob/b145afeef4007d5e485b7eab8220a20fb5ef24c0/demo/audio/client/results.js#L32)):*
<details><summary><i>Previously, asross (Andrew Ross) wrote…</i></summary><blockquote>

The popup alone won't suffice, because we need to make JSON requests to the server for the performance data we want to plot. Unfortunately but for good reasons, the credentials the user types into the page aren't accessible from Javascript. Any thoughts on how to cleanly circumvent that problem, other than using a more fully-featured authentication system?
</blockquote></details>

Does the Oauth code in the emoji demo not work for this case?
